ibis/backends/base/sql/compiler/translator.py

```diff
@@ -331,7 +327,11 @@ def _bucket(expr):
         stmt = stmt.when(cmp(op.buckets[-1], op.arg), bucket_id)
         bucket_id += 1
 
-    return stmt.end().name(expr._name)
+    result = stmt.end()
+    if expr.has_name():
+        result = result.name(expr.get_name())
```

Comment:
Looks like another place where we have new code that isn't tested.

Reply:
Breaks tests if I remove the has_name() check.